### PR TITLE
NamespaceNoLeadingWhitespaceFixer - remove need for ctype extension

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: Symfony/CS/Fixer/Symfony/NamespaceNoLeadingWhitespaceFixer.php [hunk 1]
```diff
@@ -72,6 +72,12 @@ public function getDescription()
...
     private static function endsWithWhitespace($str)
     {
-        return '' !== $str && ctype_space(substr($str, -1));
+        if ('' === $str) {
+            return false;
+        }
+
+        $last = substr($str, -1);
+
+        return $last !== trim($last);
```
Comment (gharlan) [inline Symfony/CS/Fixer/Symfony/NamespaceNoLeadingWhitespaceFixer.php]: Could be "simplified" to `return '' === trim($last);`.

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/NamespaceNoLeadingWhitespaceFixer.php]: good point ;)

**Comment by GrahamCampbell** [comment]
:+1: though i think this should go to 1.6 though given that it's a bug fix for people without this extension.

**Comment by keradus** [comment]
I disagree it is a bugfix. This lowers the requirements, but it wont fix anything.

**Comment by GrahamCampbell** [comment]
> but it wont fix anything.

It will fix fatal errors for people.

File: Symfony/CS/Fixer/Symfony/NamespaceNoLeadingWhitespaceFixer.php [hunk 2]
```diff
@@ -72,6 +72,10 @@ public function getDescription()
 
     private static function endsWithWhitespace($str)
     {
-        return '' !== $str && ctype_space(substr($str, -1));
+        if ('' === $str) {
+            return false;
+        }
+
+        return '' === trim(substr($str, -1));
```
Comment (gharlan) [inline Symfony/CS/Fixer/Symfony/NamespaceNoLeadingWhitespaceFixer.php]: Don't know if it is better, but the check is possible as one liner:

```php
    private static function endsWithWhitespace($str)
    {
        return $str !== rtrim($str);
    }
```

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/NamespaceNoLeadingWhitespaceFixer.php]: I thouth about that - but here you will trim all whitespaces on the right side.
On my PR you will check only one char.

**Comment by gharlan** [comment]
:+1:

**Comment by GrahamCampbell** [comment]
@keradus You have assigned this to the wrong milestone?

**Comment by keradus** [comment]
Yes, my bad, thank you. I will release 1.6.1 tonight.

**Pull request merged**